homeassistant/components/sensor/rejseplanen.py

```diff
@@ -0,0 +1,226 @@
+    """The Class for handling the data retrieval."""
+
+    def __init__(self, stop_id, route, direction, departure_type):
+        """Initialize the data object."""
+        self.stop_id = stop_id
+        self.route = route
+        self.direction = direction
+        self.departure_type = departure_type
+        self.info = [{ATTR_DUE_IN: 'n/a',
+                      ATTR_DUE_AT: 'n/a',
+                      ATTR_TYPE: 'n/a',
+                      ATTR_ROUTE: self.route,
+                      ATTR_DIRECTION: 'n/a',
+                      ATTR_STOP_NAME: 'n/a'}]
+
+    def update(self):
+        """Get the latest data from rejseplanen."""
+        import rjpl
+        self.info = []
+
+        try:
+            results = rjpl.departureBoard(int(self.stop_id), timeout=5)
+        except RuntimeError:
+            _LOGGER.debug("No departures returned from API call.")
+            self.info = [{ATTR_DUE_IN: 'n/a',
+                          ATTR_DUE_AT: 'n/a',
+                          ATTR_TYPE: 'n/a',
+                          ATTR_ROUTE: self.route,
+                          ATTR_DIRECTION: 'n/a',
+                          ATTR_STOP_NAME: 'n/a'}]
+            return
+
+        # Filter result
+        results = [d for d in results if 'cancelled' not in d]
+        if self.route:
+            results = [d for d in results if d['name'] in self.route]
+        if self.direction:
+            results = [d for d in results if d['direction'] in self.direction]
+        if self.departure_type:
+            results = [d for d in results if d['type'] in self.departure_type]
+
+        for item in results:
+            route = item.get('name')
+
+            due_at_date = item.get('rtDate')
+            due_at_time = item.get('rtTime')
+
+            if due_at_date is None:
+                due_at_date = item.get('date')  # Scheduled date
+            if due_at_time is None:
+                due_at_time = item.get('time')  # Scheduled time
+
+            if (due_at_date is not None and
+                    due_at_time is not None and
+                    route is not None):
+                due_at = due_at_date + " " + due_at_time
+
+                departure_data = {ATTR_DUE_IN: due_in_minutes(due_at),
+                                  ATTR_DUE_AT: due_at,
+                                  ATTR_TYPE: item.get('type'),
+                                  ATTR_ROUTE: route,
+                                  ATTR_DIRECTION: item.get('direction'),
+                                  ATTR_STOP_NAME: item.get('stop')}
+                self.info.append(departure_data)
+
+        if not self.info:
+            _LOGGER.debug("No departures with given parameters.")
+            self.info = [{ATTR_DUE_IN: 'n/a',
+                          ATTR_DUE_AT: 'n/a',
+                          ATTR_TYPE: 'n/a',
+                          ATTR_ROUTE: self.route,
+                          ATTR_DIRECTION: 'n/a',
+                          ATTR_STOP_NAME: 'n/a'}]
```

Comment:
You make this identical assignment three times, please move it to a helper method.

Reply:
Done. Testing new code now.